meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -1,17 +1,144 @@
+setenv set_blkcnt 'setexpr blkcnt ${filesize} + 0x1ff && setexpr blkcnt ${blkcnt} / 0x200'
+setenv update_image 'run set_blkcnt && mmc dev ${devnum} ${uboot_hwpart} && mmc write ${loadaddr} ${start_blk} ${blkcnt}'
+
+# Import uEnv.txt
+run bootcmd_otenv
+# Get deployment sysroot absolute path
+run bootcmd_getroot
+# Check state of SECONDARY_BOOT bit
+imx_secondary_boot
+# Check if board is on closed state
+imx_is_closed
+
+
+# Device what storage we should use for update-related vars
+# If board is closed - RPMB on eMMC
+# If boars is open - uboot environment
+if test -n "${board_is_closed}"; then
+	echo "Using fiovb"
+else
+	echo "Using ubootenv"
+	setenv fiovb.bootcount "${bootcount}"
+	setenv fiovb.rollback "${rollback}"
+	setenv fiovb.upgrade_available "${upgrade_available}"
+	setenv fiovb.bootupgrade_available "${bootupgrade_available}"
+fi
+
+
+# Handle boot firmware updates
+if test "${fiovb.is_secondary_boot}" = "0"; then
+	if test -n "${board_is_closed}"; then
+		# FIO bootcount, upgrade_available and rollback initialization
+		mmc rescan
+		fiovb init ${devnum}
+
+		fiovb read_pvalue bootcount 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootcount 0; fi
+		fiovb read_pvalue rollback 4
+		if test ! $? -eq 0; then fiovb write_pvalue rollback 0; fi
+		fiovb read_pvalue upgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue upgrade_available 0; fi
+		fiovb read_pvalue bootupgrade_available 4
+		if test ! $? -eq 0; then fiovb write_pvalue bootupgrade_available 0; fi
+	fi
+
+	# Only update bootcount when upgrade_available is set
+	if test "${fiovb.upgrade_available}" = "1"; then
+		setexpr bootcount ${fiovb.bootcount} + 1;
+		if test -n "${board_is_closed}"; then
+			fiovb write_pvalue bootcount ${bootcount};
+		fi;
+
+		# If bootcount is above bootlimit, trigger rollback
+		# if not - try to boot secondary boot path for boot firmware validation
+		if test ${bootcount} -gt ${bootlimit}; then
+			echo "FIO: doing rollback..."
+			if test -n "${board_is_closed}"; then
+				fiovb write_pvalue rollback 1;
+				fiovb write_pvalue upgrade_available 0;
+				fiovb write_pvalue bootupgrade_available 0;
+			else
+				setenv rollback 1;
+				setenv upgrade_available 0;
+				setenv bootupgrade_available 0;
+			fi;
+		else
+			if test "${fiovb.bootupgrade_available}" = "1"; then
+				imx_secondary_boot 1;
+				echo "FIO: updating secondary boot images from ${ostree_root} ..."
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader_s_image}"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk "${bootloader_s}"
+				run load_image
+				run update_image
+
+				setenv image_path "${ostree_root}/usr/lib/firmware/${bootloader2_s_image}"
+				echo "FIO: Writing ${image_path} ..."
+				setenv start_blk "${bootloader2_s}"
+				run load_image
+				run update_image
+
+				run saveenv_mmc
+				echo "FIO: doing warm reset to boot into secondary boot path..."
+				imx_warm_reset
```

Comment:
I think this is only defined for `imx8m` currently?

Reply:
https://github.com/foundriesio/u-boot/blob/2020.04%2Bfio/arch/arm/mach-imx/imx_secondary_boot.c#L98

Reply:
yes, I've stumbled upon that when doing testing of imx6ullevk

I've pulled in the same patch I sent to TF-A ML (tested also on iMX8MMEVK)
https://github.com/foundriesio/meta-lmp/pull/352